### ISSUE TITLE
What is the status of this project?  Is it still active?

### ISSUE THREAD

**Issue by TechSavvySam** [JanKallman/EPPlus#609]
I'd like to respectfully address the elephant in the room here.  I'm sure a lot of people have this question and for some reason it has not been asked here or the question has been deleted.

From my perspective it appears that this project is stalled.  I say this because:
1) The last new release was 7 months ago
2) There are 370 open issues
3) Most importantly, there are 70 open pull requests.  I assume these are where users have taken the effort to track down and fix an issue, yet those requests are just languishing.

My request for me and the community of users is:
1) Let us know what's going on
2) Inform us of your plan moving forward
3) Ask for help--I suspect this is tough for you, as it is for many of us here.

Regards,
Sam

**Comment by xin-lai** [comment]
If the author cannot support it, should we support it together?

**Comment by PrzemyslawKlys** [comment]
@xin-lai how do you propose to do that? Without the author, there's no way to update it here. Someone trusted would need to fork it and everyone should move to the fork if author doesn't want to continue.

**Comment by TechSavvySam** [comment]
When I get done with the urgent stuff I'm working on now, I'm going to investigate using Open XML SDK.  If that works well, then I think it kind of signals the oncoming demise of any open source solutions to manipulate XLSX files.  

IMO, once Microsoft provides an open source solution, newly built applications will gravitate toward that solution and interest in other solutions drops.

**Comment by mason-mcglothlin** [comment]
@TechSavvySam  In my experience, Open XML SDK was quite a bit more complicated than EPPlus. Plus, we have code that depends on EPPlus.

I share your concerns about this being a dead project. I tried to contribute a fix in June last year (#475) but haven't heard anything about when it's going to get released. My company ended up forking the code and creating our own NuGet package on our private server, but that's obviously not an ideal scenario.

**Comment by mason-mcglothlin** [comment]
Jan Källman hasn't contributed anything since 2018. I hope he's doing well! But if he no longer has the time to contribute to this project, perhaps he should begin searching for someone else to take over so that we don't have to go through the pain of forking the project and letting everyone know the new location.

**Comment by TechSavvySam** [comment]
Agree--I hope Jan is OK as well.  That was my first concern.  My hope is Jan just got busy making money and hasn't had time to devote to the project.

**Comment by xin-lai** [comment]
I hope so.
EPPlus is very good,I wrote a simpler import and export library based on it。https://github.com/dotnetcore/Magicodes.IE

**Comment by mason-mcglothlin** [comment]
Jan shows activity on LinkedIn within the past 5 months. I've sent him a message asking him to come comment on this issue.

**Comment by PrzemyslawKlys** [comment]
So did I, so let's not message him everyone now :-)

**Comment by mason-mcglothlin** [comment]
I heard back from Jan on LinkedIn. He indicated that he plans to provide some information on this thread within a few hours. Sounds like some things around here will change, and it sounds like it may be for the better.

**Comment by JanKallman** [comment]
Sorry for my absence. We have been working hard with the new version EPPlus 5.0 that will be announced soon.

To give you the background:

Over the last years it has become obvious to Mats and me that we can not continue to maintain EPPlus as a hobby/sparetime project. About a year ago we decided to start a company, EPPlus Software, to enble us to work full time with EPPlus. We have therefore taken the codebase through an IP review to ensure that the code is the intellectual property of our company. This IP review is now completed.

In the new version the license will change from LGPL to a dual license model (one paid commercial license with support included and one for noncommercial use which will be free). The refactored source code for the new version will move to a new github repo and future development will still be done publicly with the community under the new license model.

EPPlus 5 will come with some new features, which includes Themes, Filters, support for async/await on io-operations and a lot of improvements for chart styling (Office 2016 styling) and more. The sample project has been redesigned and extended.

During the coming weeks we will launch a website where commercial licenses can be purchased and publish EPPlus 5-beta on Nuget. When this happens we will also release further details regarding pricing, license terms for the commercial license, etc.

**Comment by TechSavvySam** [comment]
Many thanks for the update Jan!

**Comment by rlightner** [comment]
Wow, that went from "Hey this project is dead, and has a lot of outstanding bugs" to "We're gonna switch up the licensing model and start charging." super quick.

How about you merge the pull requests before the switch?

**Comment by fuzzzerd** [comment]
> Wow, that went from "Hey this project is dead, and has a lot of outstanding bugs" to "We're gonna switch up the licensing model and start charging." super quick.

I have to agree! There have been other projects that have done similar, and while I understand the maintainers desire to make money off their work, as a user it still leaves a sour taste when a project uses an open source license to gain traction and then switch to a paid model. 

> How about you merge the pull requests before the switch? 

In my opinion, that would certainly go a long way toward alleviating some of my angst with the license switch.

**Comment by JanKallman** [comment]
> Wow, that went from "Hey this project is dead, and has a lot of outstanding bugs" to "We're gonna switch up the licensing model and start charging." super quick.
> How about you merge the pull requests before the switch?

We can't merge any pull request into the new version, as the pull request are commited into the LGPL version. Merging all pull requests into EPPlus 4.5.3.2 AND validate that they work properly is a massiv task and nothing that we will do. You can of course do it yourself. EPPlus version 4.5.3.2 is still LGPL and can be forked and maintained by anyone who is up to the challange.
I have worked with EPPlus for more than 10 years without funding, and it has now become unsustaiable for me. Rather than letting this project stall, we move it into a product that can be financed in a sustainable way and live on.

**Comment by swmal** [comment]
We are providing a more comprehensive explanation on this topic here:

[https://www.epplussoftware.com/Home/LgplToPolyform](https://www.epplussoftware.com/Home/LgplToPolyform)

EPPlus 5 is now available as a beta on Nuget and we are fully focused on making it the - by far - best version of EPPlus so far.

@rlightner , @fuzzzerd - EPPlus 5 is not EPPlus 4 with a new license, there is a lot of [new features](https://www.epplussoftware.com/Developers/Features) and some parts of the code have been rewritten (mainly the cellstore). We are continuing to fix known bugs/issues in our [new gitrepo](https://github.com/EPPlusSoftware/EPPlus) during the beta period.

**Comment by PrzemyslawKlys** [comment]
Can you explain what does it mean non-commercial? For example does it mean I can use it within company for different needs as long as its not being sold? Or does it mean it can’t be used within companies at all and can be used for hobby projects only?

**Comment by swmal** [comment]
Hi @PrzemyslawKlys ,

It can definately be used within a commercial company with the commercial license we are offering on our website https://epplussoftware.com/LicenseOverview

I cannot give you detailed legal advice, but here is what the license states regarding free usage of EPPlus 5 and higher :

### Noncommercial Purposes
Any noncommercial purpose is a permitted purpose.

### Personal Uses
Personal use for research, experiment, and testing for the benefit of public knowledge, personal study, private entertainment, hobby projects, amateur pursuits, or religious observance, without any anticipated commercial application, is use for a permitted purpose.

### Noncommercial Organizations
Use by any charitable organization, educational institution, public research organization, public safety or health organization, environmental protection organization, or government institution is use for a permitted purpose regardless of the source of funding or obligations resulting from the funding.

The full license text is available at https://polyformproject.org/licenses/noncommercial/1.0.0/

Best regards,

Mats

**Comment by PrzemyslawKlys** [comment]
Hi @swmal,

I did read the license but like all licenses, it's a bit unclear. To me non-commercial means I can't make money directly with the product - for example selling Excel Creation software. But if that product is helping me in my business to create reports for IT Managers I believe this is non-commercial. 

However, I would prefer to be sure how owners think about this - and what is their intention. 

With regards,
Przemek

**Comment by swmal** [comment]
I would personally think of it this way: things that somehow supports a commercial business directly or indirectly would still qualify as a commercial purpose. I will reach out to our legal advisor and see if I can clarify this further.

Regarding our intention: we have discussed several different types of licenses during the setup of our company. We went with this license because it permits free usage for personal projects and non-profital organizations, whilst allowing us to continue to keep the openness of the project. The use case you are describing is probably the most common usage of EPPlus within commercial companies. We hope that as many as possible of these companies see the value in that we are trying to build a sustainable business around EPPlus so we can work full time with developing it further, provide support and fix issues.

All the best,

Mats

**Comment by gregorylyons** [comment]
The approach you've taken, including the radio silence for months and discarding many PRs, makes it very difficult to trust that you will behave in a fair and reasonable manner with the commercial version.

**Comment by swmal** [comment]
@gregorylyons There are always things you could have been done differently in retrospect. We have, to the best of our capabilities, tried to create a more solid ground for EPPlus and have communicated it when we thought the time was right. 

We were not happy with the situation around the LGPL version either, that is basically why we have worked hard to change it. Replicating a vast part of Excels functionality with your bare hands in C# whilst having your software downloaded millions of time per year is in the end not a suitable hobby project for two persons, at least that is how we feel after these ten years. We are fully confident that we now can deliver a product of much higher quality, but it is up to you to believe in us or not. 

Not sure exactly what you mean with "fair and responsible manner", but you can follow our work and progress at a very detailed level in our [new repo for EPPlus 5](https://github.com/EPPlusSoftware/EPPlus)

**Comment by jm009** [comment]
1.) Thank you, thank you, thank you, for creating EPPlus and bringing it to the state where it is.

2.) From my understanding (I may be wrong on this, but that is my knowledge level)  Open XML SDK more or less just helps you in managing the XML stuff, but keeps you alone with the file structure.
And that is, how Microsoft works - worked so often in the past: Create some really bad Software, make the most possible money out of it. I would claim, they created Open XML SDK only to give the impression, they care about open file formats.
See the story about passing OOXML as a standard. I would claim, this is one of the stories, where Microsoft used all its (financial) power, to suffocate open source software as much as they could.

3.) I would never use Excel, Open XML SDK or EPPlus if not my employer would ask me to do it and pay me for it.
And that is the reason, EPPlus is none of the projects, I would like to support in my spare time.

4.) The only reason that brings me to support EPPlus, is, because I have to use it for work, and from what I see it is a much cleaner, much more useful solution than Open XML SDK.
